### ISSUE TITLE
Stop enforcing types on methods that do not really need it. 

### ISSUE THREAD

**Issue by lrnv** [USU-Analytics-Solution-Center/Bruno.jl#69]
Here : https://github.com/USU-Analytics-Solution-Center/Bruno.jl/blob/ebc4f7493469b9ae55b5a2fabc9d407933ced9e6/src/DataGeneration/DataGeneration.jl#L49

you have a non-necessary `::Any`. 

More generally, about type enforcement on methods, I am not sure if you understood correctly how the system works. Basically, using type annotation in function signatures like 

```
f1(x::T)
```

will simply ensure that `x<:T`. In a lot of cases, this is what you want : ensure that the parameter is a `Widget`, or that it is an `EuroCallOption`. Those annotations do not bring speed to the code or anything, they just make the dispatching system work correclty, which is exactly what we want.

Generally speaking, you should refrain on using them as much as possible (except of course when it IS useful for dispatch, such as dispatching the `price!` function correctly on the asset and pricing model), because they make your code less usable by others. 

Let me explain what i mean by the `Commodity` constructor exemple: 

```
function Commodity(price::Real; name = "", volatility)
    prices = [price]
    Commodity(;
        prices = prices,
        name = name,
        volatility = volatility,
        timesteps_per_period = 0
    )
end
```

Is the `price::Real` constraint on the Commodity really necessary ? Your code will work exactly the same without it. With it, I cannot define a commodity with a non-real price, which someone else, someday, might want to do. 

Same thing goes for all of these annotations : You may take some time to think carefully about all of them, and remove any that is not needed for dispatch reasons.

**Comment by mitchellpound** [comment]
Thanks for helping us see that many of the type annotations were not useful for dispatch!
The annotations for the `Commodity` constructor (and similarly for the `Stock` and `Bond` constructors) in this case are being used for dispatch since they allow the user to construct the `Widget` with a single number as opposed to using a `Vector{<:Number}`. Without the type annotation here, when a user makes a `Commodity` with a vector such as in the example

```
julia> Commodity([25, 40, 30, 30])
Commodity{Vector{Int64}}([[25, 40, 30, 30]], "", 0, 0.3
```
so the `Commodity.prices` field is a vector of vectors. 

I also didn't take out the type annotations in the `BackTest` module (yet) since they more necessary to make sure that the user doesn't get surprised by inexact errors coming from using `<:Integer` data types. To take these out require a lot of time and careful effort. Since we are actively discussing ways to make that module more robust to user input and to better use dispatch in the simulation environment in (near) future versions, I'm not sure it would be the best investment of time right now.

**Comment by mitchellpound** [comment]
Fixed with PR #72.
